### PR TITLE
Add Google and Yandex Maps API key support / Добавить поддержку API ключа для сервисов Google и Yandex карт

### PR DESCRIPTION
**Pull request by kabalin**
### Google

Addressing issue #227 

This patchset allows to define API key in configuration file using the setting:
```
    publicApiKeys: {
        googleMaps: '<YOUR_API_KEY>'
    },
```
Alternatively, key can be supplied using env var `GOOGLE_MAPS_API_KEY`.

**Changes:**
* Layer plugin now located at `public/js/lib/leaflet/plugins/Google.js`.
* This is the latest state of Google layer plugin before author removed it from repo (shramov/leaflet-plugins@a7228ba).
* [`Leaflet.GridLayer.GoogleMutant`](https://gitlab.com/IvanSanchez/Leaflet.GridLayer.GoogleMutant) is suggested as replacement, but it seems introduce flickering issue on zoom (plugin author refers to this as "known issue"), so not considered as replacement for now, but if one day we decide to use it, developer just needs to copy it to `public/js/lib/leaflet/plugins/Google.js`, it will work with no other changes required.
* Former `public/js/lib/leaflet/extends/L.Google.js` is now a "loader".

**Query changes:**
```
- //maps.googleapis.com/maps/api/js?v=3.11&sensor=false&region=RU
+//maps.googleapis.com/maps/api/js?v=weekly&region=RU
```
`sensor=false` - no longer needed ([read more](https://developers.google.com/maps/documentation/javascript/error-messages#sensor-not-required))
 `v=weekly` - use weekly update channel.

**Note for API key maintainer:**
* It is adviced to restrict the key to HTTP referrer (`*.pastvu.com`) and Maps JavaScript API service, for more details see https://developers.google.com/maps/documentation/javascript/get-api-key#restrict_key.
* It should be also possible to cap usage if needed (and/or set alerts), for more information see https://developers.google.com/maps/faq#usage_cap

### Yandex

Addressing issue #251 

This patchset allows to define API key in configuration file using the setting:
```
    publicApiKeys: {
        yandexMaps: '<YOUR_API_KEY>'
    },
```
Alternatively, key can be supplied using env var `YANDEX_MAPS_API_KEY`.

**Changes:**
* Layer plugin is now located at `public/js/lib/leaflet/plugins/Yandex.js` (in case of upstream update, just replace it with newer version).
* Former `public/js/lib/leaflet/extends/L.Yandex.js` is now a "loader".
* Uses L.Yandex 3.3.2 (latest release retrieved from https://github.com/shramov/leaflet-plugins/)
* Loader overrides some Yandex map config options, namely disables point info popup and link to open in Yandex maps.

**Query changes:**
```
- //api-maps.yandex.ru/2.0/?load=package.map&mode=release&lang=ru-RU
+//api-maps.yandex.ru/2.1/?mode=release&lang=ru_RU
```
Use current API (`2.1`)
No need to restrict package (it is already optimised to load required parts on demand)
`mode=relese` - load minified version
Use correct region notation (`ru_RU`)
More [info on params](https://yandex.com/dev/maps/jsapi/doc/2.1/dg/concepts/load.html/#load__param).

**Note for API key maintainer:**
* It is adviced to restrict the key to HTTP referrer (`*.pastvu.com`), for more details see https://yandex.ru/dev/maps/geosearch/doc/concepts/limit.html/?lang=ru

### Other

* Both new plugins are tested against #232 issue and fix was added to each plugin as a separate commit (suprisingly Yandex layer still has the issue, may be I will send a patch to author). The case was explained in details here: https://github.com/PastVu/pastvu/pull/232#issuecomment-723509380
* Minor change was made in map related modules to use `leaflet-extends` requirejs path alias instead of relative path (b300066).
* Staging deployment and testing is adviced before merging 🙂

### CODE REVIEW

**Comment by kabalin** [comment]
Подумал я тут, надо на [Leaflet.GridLayer.GoogleMutant](https://gitlab.com/IvanSanchez/Leaflet.GridLayer.GoogleMutant) гугловский слой менять. Мерцанием на стыках плиток, , который изначально меня отпугнул, можно пренебречь (почти незаметно), а вот проблему что точки не "прыгают" при зуме этот плагин решает (приблизите тут одну из точек https://ivansanchez.gitlab.io/Leaflet.GridLayer.GoogleMutant/demo.html и сравните как ведут себя точки на продакшене когда включен слой Google). Завтра добавлю патч.

File: public/js/lib/leaflet/extends/L.Google.js [hunk 1]
```diff
@@ -1,185 +1,12 @@
...
-                }
-            }
-            L.Google.asyncWait = [];
-        }
-    })(window.google, L);
+define(['Params', 'leaflet', 'leaflet-plugins/Google'], function (P, L) {
+    let keyParam = P.settings.publicApiKeys.googleMaps.length ? '&key=' + P.settings.publicApiKeys.googleMaps : '';
+    let url = (location.protocol || 'http:') + '//maps.googleapis.com/maps/api/js?v=weekly&region=RU' + keyParam;
+    // Load Google Maps API library asynchronously.
+    require([ 'async!' + url]);
```
Comment (kabalin) [inline public/js/lib/leaflet/extends/L.Google.js:9]: Note for reviewer: `L.Google` got functionality to wait for libs availability using promise, so (unlike in Yandex case) we don't need to do anything special here apart of request libs loading and return layer class.

**Review by klimashkin — APPROVED** [review]

**Comment by aeifn** [comment]
@kabalin, спасибо! отправлено в прод!

**Comment by aeifn** [comment]
В проде всплыла ошибка 
image

**Comment by aeifn** [comment]
Починилось пересборкой образов.

**Pull request merged**